src/common/component-utils/utils.ts

```diff
@@ -1,23 +1,49 @@
-import { Children, FC, ReactElement, ReactNode } from 'react'
+import { Children, FC, ForwardRefExoticComponent, isValidElement, JSXElementConstructor, ReactNode } from 'react'
 import './array.polyfill.flat' // for Mobile Safari 11
 
-export function findComponent(nodes: ReactNode = [], componentType: FC): ReactElement | undefined {
-    const elements = Children.toArray(nodes) as ReactElement[]
-    return elements.find(({ type }) => type === componentType) || null
-}
+const componentName =
+  (component: FC<any> | JSXElementConstructor<any> | ForwardRefExoticComponent<any> | string): string | undefined => {
+      switch (typeof component) {
+          case 'string':
+              return component
+          case 'function':
+              return component.name
+          case 'symbol':
+              return (component as any).displayName
+          default:
+              return undefined
+      }
+  }
 
-export function excludeComponent(nodes: ReactNode = [], componentType: FC): ReactElement[] {
-    const elements = Children.toArray(nodes) as ReactElement[]
-    return elements.filter(({ type }) => type !== componentType)
-}
+// Returns props of a valid ReactElement
+export const elementProps = (node: ReactNode): Record<string, any> => isValidElement(node) ? node.props : {}
```

Comment:
I think the props of each component should be defined on component level, because what is the use of typrescript than when the second argument is any and one function for all components